packages/desktop/views/dashboard/drawers/dapp-config/views/ConfirmConnectionDrawer.svelte

```diff
@@ -100,15 +120,18 @@
 <DrawerTemplate title={localize(`${localeKey}.title`)} {drawerRouter}>
     <div class="w-full h-full flex flex-col space-y-6 overflow-hidden">
         {#if $sessionProposal}
+            {@const requiredNamespaces = $sessionProposal.params.requiredNamespaces}
+            {@const optionalNamespaces = $sessionProposal.params.optionalNamespaces}
+
             <DappInfo metadata={$sessionProposal.params.proposer.metadata} {verifiedState} />
 
             <div class="px-6 flex-grow overflow-hidden">
-                {#if persistedNamespaces}
+                {#if isPreferenceSelectionRequired && persistedNamespaces}
```

Comment:
If the preference selection is required you go to the Summary, but shouldn't it then go to the actual selection?